### PR TITLE
Refactor Code, Update Unit Tests, Add internal for all target to avoid dokka generate empty page

### PR DESCRIPTION
**Pull request by hieu-dd**
# Pull Request (PR) Summary

- This PR includes code refactoring, updates to unit tests, and marking certain classes as internal to prevent Dokka from generating empty documentation pages.

## Changes Description

In this PR, I have refactored parts of our codebase to improve maintainability and performance. Additionally, I have updated and added new unit tests to ensure that our code functions correctly and meets all new requirements.

To enhance our documentation process, I have marked several classes as `internal`. This change prevents Dokka, our documentation generation tool, from creating empty pages for these classes, which are not meant to be part of the public API. This will make our project documentation more concise and focused on providing value to the developers who use it.

## Related Issue(s)

- Fixes # (issue number)  // Please replace with actual issue number if applicable

## Type of Change

- [ ] Bug fix
- [ ] New feature
- [x] Documentation update
- [ ] Other (please describe)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] My PR is targeting the `main` branch
- [x] I have added necessary documentation (if appropriate)
- [x] I have added tests that prove my fix is effective or my feature works
- [x] All new and existing tests passed.

### CODE REVIEW

File: core/build.gradle.kts [hunk 1]
```diff
@@ -95,8 +95,12 @@ rootProject.plugins.withType<YarnPlugin> {
 buildkonfig {
     packageName = "dev.kryptonreborn.blockfrost.buildKonfig"
     defaultConfigs {
-        buildConfigField(BOOLEAN, "IS_CI", isCiEnv())
-        buildConfigField(STRING, "PROJECT_ID", getLocalProperty("projectId") ?: "<your project id>")
+        buildConfigField(BOOLEAN, "DISABLE_INTEGRATION_TESTS", isDisableIntegrationTests())
+        buildConfigField(
+            STRING,
+            "PROJECT_ID",
+            getLocalProperty("projectId") ?: "mainnet7fToxtolmPU20aln1LrH2brEJOwq4ZoJ",
```
Comment (iTanChi) [inline core/build.gradle.kts]: Không nên có cái này nhé

File: core/src/commonTest/kotlin/dev/kryptonreborn/blockfrost/unittest/metrics/model/MetricEndpointTest.kt [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+import dev.kryptonreborn.blockfrost.ktor.Ktor
+import dev.kryptonreborn.blockfrost.metrics.model.MetricEndpoint
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class MetricEndpointTest {
+    @Test
+    fun testDeserialization() {
+        val json =
```
Comment (iTanChi) [inline core/src/commonTest/kotlin/dev/kryptonreborn/blockfrost/unittest/metrics/model/MetricEndpointTest.kt]: mấy cái json này sao ko đưa vào resource nhỉ

File: core/build.gradle.kts (hunk 1 above)
Comment (hieu-dd) [reply core/build.gradle.kts]: quên để remove đi

File: core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/BlockFrostClient.kt [hunk 3]
```diff
@@ -1,20 +1,36 @@
...
-    private val metricsApi: MetricsApi by lazy { MetricsApi(httpClient) }
-    private val cardanoAccountsApi: CardanoAccountsApi by lazy { CardanoAccountsApi(httpClient) }
+class BlockFrostClient {
+    private val httpClient: HttpClient
+    private val healthApi: HealthApi
+    private val metricsApi: MetricsApi
+    private val cardanoAccountsApi: CardanoAccountsApi
+
+    constructor(blockfrostConfig: BlockfrostConfig) {
+        this.httpClient = Ktor.httpClient(blockfrostConfig)
```
Comment (iTanChi) [inline core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/BlockFrostClient.kt]: gọi sang internal constructor?

**Review by iTanChi — APPROVED** [review]

**Pull request merged**